### ISSUE TITLE
Organizing DeepChem Tutorials into a Cogent Sequence

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1137]
Thanks to great effort from our contributors, we have a good number of deepchem tutorials (https://deepchem.io/docs/notebooks/index.html) with more coming on the way. These tutorials provide a great resource for newcomers to DeepChem and to the field of molecular machine learning broadly. However, they're a bit dis-organized right now. It's not entirely clear where newcomers should start.

How can we organize our tutorials to help newcomers on-board better? It might make sense to mark a few tutorials as beginner overviews, and the rest as advanced case-study or API reference material.

I think it might also make sense to have one entry-point tutorial that provides a broad introduction to the library and its uses. If this sounds interesting, will start a new issue to coordinate that tutorial.

**Comment by lilleswing** [comment]
The current HTML lives here.

https://github.com/deepchem/deepchem.io/blob/master/website/docs/notebooks/index.html

A PR to the deepchem.io website would be great!

**Comment by pvskand** [comment]
@rbharath, @lilleswing,  this is a great idea indeed. Infact when I first saw the list of notebooks it was quite confusing for me.
Some thoughts on the current organization:
- In `Application Tutorials`, the first two tutorials i.e Graph Convs and Multitask Nets are good ones to start to know deepchem.
- Since we have few more Dataset tutorials (the splitters and the NumpyDataset) and few more coming soon like the DiskDataset Tutorial, I would suggest to have a separate section for Dataset Tutorials.
- The name `Machine Learning Methods Tutorials` is very vague or rather let me put it this way, some of the Application tutorials also do come under Machine Learning Methods. What we can do instead is have a separate column for Vision tasks, this way it would separate the core deepchem tasks from the vision tasks.
- Since there are few `TensorGraph` and `TensorBoard` tutorials we can have them in one separate section.
So summing up the above points we can have the following structure:

  - Dataset Tutorials
  - Beginner Tutorials for DeepChem [which include Graph Convs and Multitask Nets.]
  - Advance case-study with DeepChem
  - Vision Tutorials
  - TensorBoard and TensorGraph Tutorials

Please let me know your thoughts on this.

**Comment by lilleswing** [comment]
@pvskand what tutorials do you declare "Vision" tutorials?

**Comment by pvskand** [comment]
@lilleswing, `MNIST with deepchem` and `MNIST GAN` would fall under this category. Well I would say we can modify this to Vision & AI Tutorials and include A3C in this category as well.

**Comment by peastman** [comment]
The MNIST GAN tutorial isn't really about computer vision.  It's a tutorial on how to create GANs, and just happens to use MNIST as an example.

**Comment by pvskand** [comment]
Aah.. I see. So even the `MNIST with deepchem` is basically a classification tutorial (that uses MNIST). In that case I would suggest to have a section as `Machine Learning Tutorials` and subsections would cover `GANS` [with MNIST GAN and Conditional GAN in it] and Classification Tutorial consisting of MNIST with deepchem tutorial.

**Comment by patrickhop** [comment]
I'd love to see the tutorials more organized, and with a more thoughtful progression. Maybe something like an easy / medium / hard progression?

**Comment by pvskand** [comment]
@patrickhop, that's a good idea. We can have tags for each of the tutorials explaining the difficulty level of the tutorial. But having the tutorials organized only on the basis of easy/medium/hard would not be an effective organization. What do you say?

**Comment by patrickhop** [comment]
@pvskand I agree that we shouldn't explicitly bucket on the basis of difficulty, but it's important that we have tutorials that target each level. I think we're on the same page here.

**Comment by rbharath** [comment]
Closing this old issue. The tutorials currently need a refresh which is something I'm working on.
